src/webhook/mutation/pod/pod_mutator.go

```diff
@@ -0,0 +1,152 @@
+package pod
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/src/webhook"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// AddPodMutationWebhookToManager adds the Webhook server to the Manager
+func AddPodMutationWebhookToManager(mgr manager.Manager, ns string) error {
+	podName := os.Getenv("POD_NAME")
+	if podName == "" {
+		log.Info("no Pod name set for webhook container")
+	}
+
+	if err := registerInjectEndpoint(mgr, ns, podName); err != nil {
+		return err
+	}
+	registerLivezEndpoint(mgr)
+	return nil
+}
+
+// podMutatorWebhook injects the OneAgent into Pods
+type podMutatorWebhook struct {
```

Comment:
None of `podMutatorWebhook` methods need pointer receiver, please change them to value receivers.

Reply:
I think higher memory consumption won't be noticed in this case. 
I also tend to use pointer receivers but this is premature optimization and if value receiver is possible, is should be used in a first place.
Object immutability makes code safer and easier to maintain.